cmake/findDependencies.cmake

```diff
@@ -59,3 +59,30 @@ if( BUILD_WITH_CCACHE )
     endif()
 endif()
 
+set(CMAKE_REQUIRED_FLAGS "-std=c++11")
+# source: https://stackoverflow.com/questions/12530406/is-gcc-4-8-or-earlier-buggy-about-regular-expressions/41186162#41186162
+check_cxx_source_compiles(
+  "#include <regex>
+#if __cplusplus >= 201103L &&                             \
+    (!defined(__GLIBCXX__) || (__cplusplus >= 201402L) || \
+        (defined(_GLIBCXX_REGEX_DFS_QUANTIFIERS_LIMIT) || \
+         defined(_GLIBCXX_REGEX_STATE_LIMIT)           || \
+             (defined(_GLIBCXX_RELEASE)                && \
+             _GLIBCXX_RELEASE > 4)))
+int main() { return 0; }
+#else
+#error \"regex not working\"
+#endif
+"
+  BUILTIN_REGEX_WORKING
+  FAIL_REGEX '.*regex not working.*'
+  )
+set(CMAKE_REQUIRED_FLAGS "")
```

Comment:
[minor] According to the CMake doc, `CMAKE_REQUIRED_FLAGS` only affects to the `check_cxx_source_compiles` and `check_c_source_compiles` commands. Therefore, I think we do not need to clear the variable here. 

https://cmake.org/cmake/help/latest/module/CheckCXXSourceCompiles.html?highlight=checkcxxsourcecompiles

Reply:
Unfortunately it is required to clear it. I initially did not do that and the XMPSDK would suddenly no longer compiler, clearing it fixed the issue. Don't ask me why…